pkg/snapshot/restorer/restorer.go

```diff
@@ -189,19 +208,23 @@ func (r *Restorer) makeDB(snapdir string, snap *brtypes.Snapshot, commit int, sk
 	}
 	defer rc.Close()
```

Comment:
This is done too late. It should be done just after the following code:
```go
rc, err := r.store.Fetch(*snap)
	if err != nil {
		return err
	}
```
If `compressor.IsSnapshotCompressed` returns an error then defer is never setup and you have an open `ReadCloser` leaking memory.

Reply:
Agreed. I've moved `defer rc.Close()` to just after Fetch() call. Thanks